be/src/column/type_traits.h

```diff
@@ -244,6 +248,11 @@ struct ColumnTraits<int16_t> {
     using ColumnType = Int16Column;
 };
 
+template <>
+struct ColumnTraits<uint24_t> {
+    using ColumnType = UInt24Column;
```

Comment:
Why need UInt24Column?

Reply:
This is used to store old date type in storage layer.

Reply:
Vectorized engine haven't use it, don't need add a column for it, and we don't support direct upgrade from the earlier version to the latest version.

Reply:
Cause some dependent code has not been deleted completely...

I need this `ColumnTraits` to handle all type-specific code, but found that some deprecated type could not be handled by this traits. To make compiler happy, I need to implement traits for these deprecated types.

BTW, is there any plan to remove these deprecated type and code?

Reply:
BTW, these types are really confusing:
- decimal, decimalV2, decimalV3, 
- date, dateV2
- object(which is a bitmap), bitmap

Reply:
I decide to move uint24 and decimal12_t specialization from `ColumnTraits` to `CppColumnTraits`, to avoid confusion.

The `CppColumnTraits` is applied for infer `ColumnType` from `FieldType`, which supports legacy type `DATE` and `DECIMAL`.